homeassistant/components/climate/xiaomi_miio.py

```diff
@@ -0,0 +1,624 @@
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.components.climate import (
+    PLATFORM_SCHEMA,
+    ClimateDevice, ATTR_TARGET_TEMP_HIGH, ATTR_TARGET_TEMP_LOW)
+from homeassistant.const import (
+    TEMP_CELSIUS, ATTR_TEMPERATURE, ATTR_UNIT_OF_MEASUREMENT,
+    CONF_NAME, CONF_HOST, CONF_TOKEN, CONF_TIMEOUT)
+# from homeassistant.helpers import condition
+from homeassistant.helpers.event import (
+    async_track_state_change, async_track_time_interval)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['python-miio==0.3.2']
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['sensor']
+
+DEFAULT_TOLERANCE = 0.3
+DEFAULT_NAME = 'Mi ACpartner'
+
+DEFAULT_TIMEOUT = 10
+DEFAULT_RETRY = 3
+
+DEFAULT_MIN_TMEP = 16
+DEFAULT_MAX_TMEP = 30
+DEFAULT_STEP = 1
+
+CONF_SENSOR = 'target_sensor'
+# CONF_TARGET_TEMP = 'target_temp'
+CONF_SYNC = 'sync'
+CONF_CUSTOMIZE = 'customize'
+
+__Presets__ = {
+    "default": {
+        "description": "The Default Replacement of AC Partner",
+        "defaultMain": "AC model(10)+po+mo+wi+sw+tt",
+        "VALUE": ["po", "mo", "wi", "sw", "tt", "li"],
+        "po": {
+            "type": "switch",
+            "on": "1",
+            "off": "0"
+        },
+        "mo": {
+            "heater": "0",
+            "cooler": "1",
+            "auto": "2",
+            "dehum": "3",
+            "airSup": "4"
+        },
+        "wi": {
+            "auto": "3",
+            "1": "0",
+            "2": "1",
+            "3": "2"
+        },
+        "sw": {
+            "on": "0",
+            "off": "1"
+        },
+        "tt": "1",
+        "li": {
+            "off": "a0"
+        }
+    },
+    "0180111111": {
+        "des": "media_1",
+        "main": "0180111111pomowiswtt02"
+    },
+    "0180222221": {
+        "des": "gree_1",
+        "main": "0180222221pomowiswtt02"
+    },
+    "0100010727": {
+        "des": "gree_2",
+        "main": "0100010727pomowiswtt1100190t0t20500\
+                2102000t6t0190t0t207002000000t4wt0",
+        "off": "010001072701011101004000205002112000\
+                D04000207002000000A0",
+        "EXTRA_VALUE": ["t0t", "t6t", "t4wt"],
+        "t0t": "1",
+        "t6t": "7",
+        "t4wt": "4"
+    },
+    "0100004795": {
+        "des": "gree_8",
+        "main": "0100004795pomowiswtt0100090900005002"
+    },
+    "0180333331": {
+        "des": "haier_1",
+        "main": "0180333331pomowiswtt12"
+    },
+    "0180666661": {
+        "des": "aux_1",
+        "main": "0180666661pomowiswtt12"
+    },
+    "0180777771": {
+        "des": "chigo_1",
+        "main": "0180777771pomowiswtt12"
+    }
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+    vol.Required(CONF_SENSOR, default=None): cv.entity_id,
+    vol.Optional(CONF_CUSTOMIZE, default=None): dict,
```

Comment:
You seem to use the `_customize` member to set fan and swing modes. Why don't you just let the user configure a list of possible fan and swing modes, as for example the `mqtt` climate platform does?